client/java/src/main/java/com/linecorp/centraldogma/client/CentralDogma.java

```diff
@@ -512,10 +513,31 @@
      *     assert myValue instanceof MyType;
      *     ...
      * });}</pre>
+     *
+     * <p>Note that {@link Function} by default is executed by a blocking task executor so that you can
+     * safely call a blocking operation.
      */
     <T, U> Watcher<U> fileWatcher(String projectName, String repositoryName,
                                   Query<T> query, Function<? super T, ? extends U> function);
 
+    /**
+     * Returns a {@link Watcher} which notifies its listeners after applying the specified
+     * {@link Function} when the result of the given {@link Query} becomes available or changes. e.g:
+     * <pre>{@code
+     * Watcher<MyType> watcher = client.fileWatcher(
+     *         "foo", "bar", Query.ofJson("/baz.json"),
+     *         content -> new ObjectMapper().treeToValue(content, MyType.class));
+     *
+     * watcher.watch((revision, myValue) -> {
+     *     assert myValue instanceof MyType;
+     *     ...
+     * });}</pre>
+     *
+     * @param executor the {@link Executor} that executes the {@link Function}
+     */
+    <T, U> Watcher<U> fileWatcher(String projectName, String repositoryName,
+                                  Query<T> query, Function<? super T, ? extends U> function, Executor executor);
```

Comment:
I'm wondering if we need to provide this method that takes an additional `executor` because the client is already created with a blocking task executor. 🤔

Reply:
I thought, sometimes, users want to specify different executors depending on the query they use.

Reply:
That makes sense. 👍